### ISSUE TITLE
Support OCaml 4.10

### ISSUE THREAD

**Issue by XVilka** [BinaryAnalysisPlatform/bap#1110]
```
~/data/ocaml took 2s 
[i] ℤ opam install bap                                                                                                                                                                                                             13:05:04 
The following dependencies couldn't be met:
  - bap → bap-std (>= 1.2.0 & < 1.4.0) → ppx_jane < v0.9 → ocaml = 4.02.3
      base of this switch (use `--unlock-base' to force)
  - bap → bap-std (>= 1.2.0 & < 1.4.0) → ppx_jane < v0.9 → ppx_bench < 113.25.00 → ppx_core < 113.25.00 → ocaml < 4.03.0
      base of this switch (use `--unlock-base' to force)
  - bap → bap-std (>= 1.2.0 & < 1.4.0) → ppx_jane < v0.9 → ppx_bench < 113.25.00 → ocaml < 4.03
      base of this switch (use `--unlock-base' to force)
  - bap → bap-std (>= 1.2.0 & < 1.4.0) → ppx_jane < v0.9 → ppx_custom_printf < 113.34.00+4.03 → ppx_core < 113.34.00+4.03 → ocaml < 4.05
      base of this switch (use `--unlock-base' to force)
  - bap → bap-std (>= 1.2.0 & < 1.4.0) → ppx_jane < v0.9 → ppx_custom_printf < 113.34.00+4.03 → ocaml < 4.03
      base of this switch (use `--unlock-base' to force)
  - bap → oasis < 0.4.7 → ocaml < 4.06.0
      base of this switch (use `--unlock-base' to force)
  - bap → ocaml < 4.10.0
      base of this switch (use `--unlock-base' to force)

No solution found, exiting
~/data/ocaml took 3s 
[i] ℤ opam info ocaml                                                                                                                                                                                                              13:05:20 

<><> ocaml: information on all versions <><><><><><><><><><><><><><><><><><><><>
name                   ocaml
all-installed-versions 4.08.1 [4.08.1]  4.10.0 [ocaml-system]
all-versions           3.07  3.07+1  3.07+2  3.08.0  3.08.1  3.08.2  3.08.3  3.08.4  3.09.0  3.09.1  3.09.2  3.09.3  3.10.0  3.10.1  3.10.2  3.11.0  3.11.1  3.11.2  3.12.0  3.12.1  4.00.0  4.00.1  4.01.0  4.02.0  4.02.1  4.02.2  4.02.3
                       4.03.0  4.04.0  4.04.1  4.04.2  4.05.0  4.06.0  4.06.1  4.07.0  4.07.1  4.08.0  4.08.1  4.09.0  4.09.1  4.10.0  4.10.1  4.11.0  4.12.0

<><> Version-specific details <><><><><><><><><><><><><><><><><><><><><><><><><>
version      4.10.0
repository   default
homepage:    "https://ocaml.org"
bug-reports: "https://github.com/ocaml/ocaml/issues"
authors:     "Xavier Leroy"
             "Damien Doligez"
             "Alain Frisch"
             "Jacques Garrigue"
             "Didier Rémy"
             "Jérôme Vouillon"
maintainer:  "platform@lists.ocaml.org"
depends:     "ocaml-config"
             "ocaml-base-compiler" {= "4.10.0"} |
             "ocaml-variants" {>= "4.10.0" & < "4.10.1~"} |
             "ocaml-system" {>= "4.10.0" & < "4.10.1~"}
synopsis     The OCaml compiler (virtual package)
description  This package requires a matching implementation of OCaml,
             and polls it to initialise specific variables like `ocaml:native-dynlink`
```

Some distributions already switched to 4.10 https://repology.org/project/ocaml/versions

**Comment by ivg** [comment]
As always, it is i not about the OCaml version, rather than about Cor and its own constraints. If core v0.12.3 indeed [supports][1] OCaml 4.10.0, then we can safely add this version to our list. Otherwise it will require dropping the 4.07 version (that is currently very heavily used by many of our users) so we have to wait for at least a quarter before doing this. But certainly 4.10 will be supported in BAP 2.2.0

[1]: https://opam.ocaml.org/packages/core_kernel/core_kernel.v0.12.3/

**Comment by ivg** [comment]
fixed in #1116
